src/main/java/com/courtcanva/ccfranchise/jwts/JwtUsernameAndPasswordAuthenticationFilter.java

```diff
@@ -0,0 +1,69 @@
+package com.courtcanva.ccfranchise.jwts;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.jsonwebtoken.Jwts;
+import lombok.SneakyThrows;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.crypto.SecretKey;
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.time.LocalDate;
+import java.util.Date;
+
+
+public class JwtUsernameAndPasswordAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    private static final String LOGIN_URL = "/staff/signin";
+    private static final String BEARER = "Bearer ";
+    private final AuthenticationManager authenticationManager;
+    private final SecretKey secretKey;
+    private final JwtConfig jwtConfig;
+
+    public JwtUsernameAndPasswordAuthenticationFilter(AuthenticationManager authenticationManager, SecretKey secretKey, JwtConfig jwtConfig) {
+
+        super.setFilterProcessesUrl(LOGIN_URL);
+        this.authenticationManager = authenticationManager;
+        this.secretKey = secretKey;
+        this.jwtConfig = jwtConfig;
+
+    }
+
+    @Override
+    @SneakyThrows
+    public Authentication attemptAuthentication(HttpServletRequest request,
+                                                HttpServletResponse response) throws AuthenticationException {
+
+        UsernameAndPasswordAuthenticationRequest authenticationRequest = new ObjectMapper()
+                .readValue(request.getInputStream(), UsernameAndPasswordAuthenticationRequest.class);
```

Comment:
I think we have an easier way to get the user's name and password. you can check the `attemptAuthentication` implementation of `UsernamePasswordAuthenticationFilter` Class